### PR TITLE
irrlicht: build and install framework

### PR DESCRIPTION
**Pull request by FnControlOption**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Review by carlocab — DISMISSED** [review]
Thanks, @FnControlOption. Squashed your commits together to make it easier for @BrewTestBot to merge.

File: Formula/irrlicht.rb [hunk 1]
```diff
@@ -43,10 +44,25 @@ def install
...
 
+      extra_args = []
+
+      # Fix "Undefined symbols for architecture arm64: "_png_init_filter_functions_neon"
+      # Reported 18 Nov 2020 https://sourceforge.net/p/irrlicht/bugs/452/
+      extra_args << "GCC_PREPROCESSOR_DEFINITIONS='PNG_ARM_NEON_OPT=0'" if Hardware::CPU.arm?
+
+      xcodebuild "-project", "source/Irrlicht/MacOSX/MacOSX.xcodeproj",
+                 "-configuration", "Release",
+                 "-target", "IrrFramework",
```
Comment (carlocab) [inline Formula/irrlicht.rb:55]: Do they not have a shared library target?

Comment (FnControlOption) [reply Formula/irrlicht.rb:55]: Nope, but we could do [what the MacPorts project does](https://github.com/macports/macports-ports/blob/master/devel/irrlicht/Portfile#L71) and add a symlink:

```rb
frameworks.install "source/Irrlicht/MacOSX/build/Release/IrrFramework.framework"
ln_s frameworks/"IrrFramework.framework/Versions/A/IrrFramework", lib/"libIrrlicht.dylib"
```

Comment (carlocab) [reply Formula/irrlicht.rb:55]: Yea, may as well:
```ruby
include.install_symlink frameworks/"IrrFramework.framework/Versions/A/Headers" => "irrlicht"
lib.install_symlink frameworks/"IrrFramework.framework/Versions/A/IrrFramework" => "libIrrlicht.dylib"
```

Comment (carlocab) [reply Formula/irrlicht.rb:55]: Oh, wait, you've already installed the headers into `include`.

Comment (carlocab) [reply Formula/irrlicht.rb:55]: Cancelled the merge so that we can make your changes here.

**Review by carlocab — DISMISSED** [review]

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**